banking-microservice/src/main/java/com/banking/bankingmicroservicetask/service/BankAccountServiceImpl.java

```diff
@@ -1,76 +1,90 @@
+  @Override
+  public Mono<Void> saveAccount(BankAccountDto bankAccountDto) {
+    log.debug("Saving bank account: {}", bankAccountDto);
 
-        return bankAccountMapper.bankAccountToBankAccountDto(bankAccount);
-    }
+    BankAccount bankAccount = bankAccountMapper.bankAccountDtoToBankAccount(bankAccountDto);
 
-    @Override
-    public void deleteAccount(UUID id) {
-        BankAccount bankAccount = bankAccountRepository.findByIdAndBankAccountDeletedTimeIsNull(id).orElseThrow(NoSuchBankAccountException::new);
+    return bankAccountRepository.save(bankAccount)
+        .doOnNext(savedAccount -> log.debug("Saved bank account with id: {}", savedAccount.getId()))
+        .then();
+  }
 
-        log.debug("Deleting bank account with id: {}", id);
+  @Override
+  public Mono<BankAccountDto> getAccount(UUID id) {
+    log.debug("Fetching bank account with id: {}", id);
 
-        bankAccount.setBankAccountDeletedTime(LocalDateTime.now());
-        bankAccountRepository.save(bankAccount);
+    return bankAccountRepository.findByIdAndBankAccountDeletedTimeIsNull(id)
+        .switchIfEmpty(Mono.error(new NoSuchBankAccountException()))
+        .map(bankAccountMapper::bankAccountToBankAccountDto)
+        .doOnNext(account -> log.debug("Fetched bank account with id: {}", id));
+  }
 
-        log.debug("Deleted bank account with id: {}", id);
-    }
+  @Override
+  public Mono<Void> deleteAccount(UUID id) {
+    return bankAccountRepository.findByIdAndBankAccountDeletedTimeIsNull(id)
+        .switchIfEmpty(Mono.error(() -> {
+          throw new NoSuchBankAccountException();
+        }))
+        .flatMap(account -> {
+          account.setBankAccountDeletedTime(LocalDateTime.now());
+          return bankAccountRepository.save(account);
+        })
+        .doOnSuccess((account) -> log.debug("Deleted bank account with id: {}", id))
+        .then();
+  }
 
   @Override
-  //TODO flux
-  public List<BankAccountDto> getAll() {
+  public Flux<BankAccountDto> getAll() {
     log.debug("Fetching all bank accounts");
-    return bankAccountRepository.findAll().stream()
-        .map(bankAccountMapper::bankAccountToBankAccountDto)
-        .toList();
+    return bankAccountRepository.findAll()
+        .map(bankAccountMapper::bankAccountToBankAccountDto);
   }
 
-    @Scheduled(cron = "#{@bankAccountServiceProperties.cron}")
-    public void updateLimits() {
-        List<BankAccount> accounts = bankAccountRepository.findAll();
-        for (BankAccount account : accounts) {
-            account.setLimitServices(account.getTransactionLimitServices().getLimitSum());
-            account.setLimitGoods(account.getTransactionLimitGoods().getLimitSum());
-            bankAccountRepository.saveAll(accounts);
-        }
-    }
+  @Scheduled(cron = "#{@bankAccountServiceProperties.cron}")
+  public void updateLimits() {
+    bankAccountRepository.findAll()
+        .publishOn(Schedulers.boundedElastic())
+        .flatMap(account -> {
```

Comment:
Did you read about zipWith and switchIfEmpty?

Reply:
I've read, but transactionLimitRepository isn't a ReactiveCrudRepository, but CrudRepository, that's why I couldn't use switchIfEmpty. Now it's fixed.